packages/beacon-node/test/unit/chain/archiveStore/utils/hierarchicalLayers.test.ts

```diff
@@ -0,0 +1,168 @@
+      },
+      // An epoch after first snapshot
+      {
+        title: "slot before epoch 8",
+        slot: computeStartSlotAtEpoch(8) - 1,
+        output: {snapshotSlot: computeStartSlotAtEpoch(7), diffSlots: []},
+      },
+      {
+        title: "slot at epoch 8",
+        slot: computeStartSlotAtEpoch(8),
+        output: {snapshotSlot: computeStartSlotAtEpoch(7), diffSlots: [computeStartSlotAtEpoch(8)]},
+      },
+      {
+        title: "slot after epoch 8",
+        slot: computeStartSlotAtEpoch(8) + 1,
+        output: {snapshotSlot: computeStartSlotAtEpoch(7), diffSlots: [computeStartSlotAtEpoch(8)]},
+      },
+    ];
+
+    it.each(overlappingEpochs)("$title", ({slot, output}) => {
+      const hierarchicalLayers = HierarchicalLayers.fromString("1,3,5,7");
+
+      expect(hierarchicalLayers.getArchiveLayers(slot)).toEqual(output);
+    });
+
+    // Considering that there are 8 slots per epoch and we are using 3,5,7 epoch as layers
+    const nonOverlappingEpochs: {title: string; slot: number; output: Layers}[] = [
+      {title: "genesis slot", slot: 0, output: {snapshotSlot: 0, diffSlots: []}},
+      {title: "slot after genesis slot", slot: 5, output: {snapshotSlot: 0, diffSlots: []}},
+      {
+        title: "one slot before first diff layer",
+        slot: computeStartSlotAtEpoch(3) - 1,
+        output: {snapshotSlot: 0, diffSlots: []},
+      },
+      {
+        title: "at slot of first diff layer",
+        slot: computeStartSlotAtEpoch(3),
+        output: {snapshotSlot: 0, diffSlots: [computeStartSlotAtEpoch(3)]},
+      },
+      {
+        title: "after slot of first diff layer",
+        slot: computeStartSlotAtEpoch(3) + 1,
+        output: {snapshotSlot: 0, diffSlots: [computeStartSlotAtEpoch(3)]},
+      },
+      {
+        title: "one slot before second diff layer",
+        slot: computeStartSlotAtEpoch(5) - 1,
+        output: {snapshotSlot: 0, diffSlots: [computeStartSlotAtEpoch(3)]},
+      },
+      {
+        title: "at slot of second diff layer",
+        slot: computeStartSlotAtEpoch(5),
+        output: {snapshotSlot: 0, diffSlots: [computeStartSlotAtEpoch(5)]},
+      },
+      {
+        title: "after slot of second diff layer",
+        slot: computeStartSlotAtEpoch(5) + 1,
+        output: {snapshotSlot: 0, diffSlots: [computeStartSlotAtEpoch(5)]},
+      },
+      {
+        title: "one slot before first snapshot",
+        slot: computeStartSlotAtEpoch(7) - 1,
+        output: {snapshotSlot: 0, diffSlots: [computeStartSlotAtEpoch(5), computeStartSlotAtEpoch(6)]},
+      },
+      {
+        title: "at slot of second diff layer",
+        slot: computeStartSlotAtEpoch(7),
+        output: {snapshotSlot: computeStartSlotAtEpoch(7), diffSlots: []},
+      },
+      {
+        title: "after slot of second diff layer",
+        slot: computeStartSlotAtEpoch(7) + 1,
+        output: {snapshotSlot: computeStartSlotAtEpoch(7), diffSlots: []},
+      },
```

Comment:
Can you add a test case here for `slot = computeStartSlotAtEpoch(12)`?

This would showcase slots from every diff layers, and non-zero slot from snapshot layer are used.

Reply:
Added